include/cupla/cudaToCupla/driverTypes.hpp

```diff
@@ -68,24 +64,58 @@
-  static_cast<uint3>(                                                \
+  static_cast<cupla::uint3>(                                                \
       ::alpaka::idx::getIdx<::alpaka::Block, ::alpaka::Threads>(acc))
 
 #define gridDim                                                                \
-  static_cast<uint3>(                                                \
+  static_cast<cupla::uint3>(                                                \
       ::alpaka::workdiv::getWorkDiv<::alpaka::Grid, ::alpaka::Blocks>(acc))
 #define blockDim                                                               \
-  static_cast<uint3>(                                                \
+  static_cast<cupla::uint3>(                                                \
       ::alpaka::workdiv::getWorkDiv<::alpaka::Block, ::alpaka::Threads>(acc))
 #define elemDim                                                               \
-  static_cast<uint3>(                                                \
+  static_cast<cupla::uint3>(                                                \
       ::alpaka::workdiv::getWorkDiv<::alpaka::Thread, ::alpaka::Elems>(acc))
 
 // atomic functions
 #define atomicAdd(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Add>(acc, ppPointer, ppValue)
+#define atomicSub(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Sub>(acc, ppPointer, ppValue)
+#define atomicMin(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Min>(acc, ppPointer, ppValue)
+#define atomicMax(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Max>(acc, ppPointer, ppValue)
+#define atomicInc(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Inc>(acc, ppPointer, ppValue)
+#define atomicDec(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Dec>(acc, ppPointer, ppValue)
 #define atomicExch(ppPointer,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Exch>(acc, ppPointer, ppValue)
-
-#define uint3 ::cupla::uint3
+#define atomicCAS(ppPointer,ppCompare,ppValue) ::alpaka::atomic::atomicOp<::alpaka::atomic::op::Cas>(acc, ppPointer, ppCompare, ppValue)
+
+// recast functions
+/* defining these as inling functions will result in multiple declaration
+ * errors when using a CUDA accelerator with alpaka.
+ * Note that there may be no spaces between the macro function name and
+ * the argument parentheses. */
+namespace cupla {
+
+
+    /* no matter how ALPAKA_FN_HOST_ACC is defined, we want to inline this */
+    template< class A, class B >
+    ALPAKA_FN_NO_INLINE_HOST_ACC inline
+    B __A_as_B( A const & x )
+    {
+        static_assert( sizeof(A) == sizeof(B), "reinterpretation assumes data types of same size!" );
+        union ba { B b; A a; };
+        ba tmp;
+        tmp.a = x;
+        return tmp.b;
+    }
+
+
+} // namespace cupla
+
+#ifndef ALPAKA_ACC_GPU_CUDA_ENABLED
+#   define __int_as_float( cupla::__A_as_B< int, float > )
```

Comment:
well, I just applied it already, but why variational arguments? `A_AS_B` allows only 1 argument anyway. Furthermore the above define should find and replace all `__int_as_float` with `( cupla::__A_as_B< int, float > )` if I understand correctly, so it works exactly the same, because then instances of :

    __int_as_float(x);

will be converted to:

    ( cupla::__A_as_B< int, float > )(x)

although this is much much lower level.